src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1696,29 +1696,7 @@ internal Solution WithFrozenPartialCompilationIncludingSpecificDocument(Document
 
         internal ImmutableArray<DocumentId> GetRelatedDocumentIds(DocumentId documentId)
         {
-            var projectState = _state.GetProjectState(documentId.ProjectId);
-            if (projectState == null)
-            {
-                // this document no longer exist
-                return ImmutableArray<DocumentId>.Empty;
-            }
-
-            var documentState = projectState.DocumentStates.GetState(documentId);
-            if (documentState == null)
-            {
-                // this document no longer exist
-                return ImmutableArray<DocumentId>.Empty;
-            }
-
-            var filePath = documentState.FilePath;
-            if (string.IsNullOrEmpty(filePath))
-            {
-                // this document can't have any related document. only related document is itself.
-                return ImmutableArray.Create(documentId);
-            }
-
-            var documentIds = GetDocumentIdsWithFilePath(filePath);
-            return this.FilterDocumentIdsByLanguage(documentIds, projectState.ProjectInfo.Language);
```

Comment:
We didn't delete FilterDocumentIdsByLanguage from wherever it originally came from?